e2e/cypress/integration/bots/managing_bot_accounts_spec.js

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @bots
+
+describe('Managing bot accounts', () => {
+    let newTeam;
+    const botName = 'bot-' + Date.now();
+
+    before(() => {
+        // # Set ServiceSettings to expected values
+        const newSettings = {
+            ServiceSettings: {
+                EnableBotAccountCreation: true,
+                DisableBotsWhenOwnerIsDeactivated: true,
+            },
+            PluginSettings: {
+                Enable: true,
+                RequirePluginSignature: false,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        // # Create a test bot
+        cy.apiCreateBot(botName, 'Test Bot', 'test bot');
+
+        // # Create and visit new channel
+        cy.apiInitSetup().then(({team}) => {
+            newTeam = team;
+        });
+    });
+
+    it('MM-T1851 No option to create BOT accounts when Enable Bot Account Creation is set to False.', () => {
+        // # Visit bot config
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/integrations/bot_accounts');
+
+        // # Click 'false' to disable
+        cy.get('[data-testid="ServiceSettings.EnableBotAccountCreation"] > .col-sm-8 > :nth-child(3)').click();
+
+        // # Save
+        cy.findByTestId('saveSetting').should('be.enabled').click();
+
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // * Assert that adding bots is not possible
+        cy.get('#addBotAccount').should('not.be.visible');
+    });
+
+    it('MM-T1852 Bot creation via API is not permitted when Enable Bot Account Creation is set to False', () => {
+        // # Visit bot config
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/integrations/bot_accounts');
+
+        // # Click 'false' to disable
+        cy.get('[data-testid="ServiceSettings.EnableBotAccountCreation"] > .col-sm-8 > :nth-child(3)').click();
+
+        // # Save
+        cy.findByTestId('saveSetting').should('be.enabled').click();
```

Comment:
This failed on me due to "Save" button not enabled since no change happened on `EnableBotAccountCreation`. Previous test set it to `false` already. You might need to update config before each test case.